File: programs-basic-java/src/test/java/com/local/numbertypes/NumbersRangeTest.java
package com.local.numbertypes;

import static org.junit.jupiter.api.Assertions.assertTrue;

import java.util.Arrays;
import java.util.List;
import java.util.stream.Collectors;
import java.util.stream.LongStream;

import org.junit.jupiter.api.Test;

import lombok.extern.slf4j.Slf4j;

@Slf4j
class NumbersRangeTest {

	@Test
	void testNumberTypesInRanges() {
		// Input Range tested is : 10 - 5000
		//@formatter:off			
		Arrays.stream(NumberType.values()).forEach(enumValue -> {
			List<Long> response = LongStream.rangeClosed(1, 1000)
					.filter(value -> enumValue.getOperation().execute(value, 5))
					.boxed()
					.collect(Collectors.toList());
			//@formatter:on
			log.info("{}: {} \t {}", enumValue.getNumberType(), response.size(),
					response.stream().map(String::valueOf).collect(Collectors.joining(",")));
			assertTrue(response.size() > 0, "Expected atleast one number to be present");
		});
	}
}

File: programs-basic-kotlin/src/test/kotlin/com/local/numbertypes/NumbersRangeTest.kt
package com.local.numbertypes
import mu.KotlinLogging
import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.Test
import java.util.function.Function
import java.util.stream.Collectors
import java.util.stream.LongStream

private val logger = KotlinLogging.logger {}
class NumbersRangeTest {
	@Test
	fun testNumberTypesInRanges() {
		// Input Range tested is : 10 - 5000
		NumberType.values().forEach { numberType ->
				val response = LongStream.rangeClosed(1 , 1000)
						.filter { value : Long -> numberType.getOperation().execute(value , 5) }
						.boxed()
						.collect(Collectors.toList());
				logger.info("{}: {} \t {}" , numberType.numberType , response.size ,
						response.stream().map<String>(Function<Long , String> { obj : Long -> obj.toString() })
								.collect(Collectors.joining(",")))
				Assertions.assertTrue(response.size > 0 , "Expected atleast one number to be present")

		}
	}
}